SMHE-1441 PING functionaliteit werkt niet meer op Productie.

natasja-n opened the pull request:
set ip address from message meta data if device had no ip address

https://jenkins.fdp.osgp.cloud/job/gxf-multibranch-build/job/PR-1006/10/

natasja-n committed:
SMHE-1441 PING functionaliteit werkt niet meer op Productie.
set ip address from message meta data if device had no ip address

Signed-off-by: Natasja Nortier <natasja.nortier@alliander.com>

kroesctrl reviewed (changes requested):
Ik denk dat dit niet goed is, messagemeta wordt op ander plek al gevuld

natasja-n committed:
SMHE-1441 PING functionaliteit werkt niet meer op Productie.
set ip address from message meta data if device had no ip address

Signed-off-by: Natasja Nortier <natasja.nortier@alliander.com>

kroesctrl commented on osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/factories/DlmsConnectionHelper.java:
```diff
@@ -84,6 +89,10 @@ public void createAndHandleConnectionForDevice(
 
+    if (device.getIpAddress() == null) {
+      this.domainHelperService.setIpAddressFromMessageMetadataOrSessionProvider(
+          device, messageMetadata);
```
setIpAddressFromMessageMetadataOrSessionProvider will now be invoked twice...... Maybe we should move the if == null to the method setIpAddressFromMessageMetadataOrSessionProvider

natasja-n replied:
The null check is moved to the method and a unit test is added.

natasja-n committed:
SMHE-1441 PING functionaliteit werkt niet meer op Productie.
move check if ip address == null to DomainHelperService setIpAddressFromMessageMetadataOrSessionProvider

Signed-off-by: Natasja Nortier <natasja.nortier@alliander.com>

natasja-n committed:
SMHE-1441 PING functionaliteit werkt niet meer op Productie.
move check if ip address == null to DomainHelperService setIpAddressFromMessageMetadataOrSessionProvider

Signed-off-by: Natasja Nortier <natasja.nortier@alliander.com>

natasja-n committed:
SMHE-1441 PING functionaliteit werkt niet meer op Productie.
move check if ip address == null to DomainHelperService setIpAddressFromMessageMetadataOrSessionProvider

Signed-off-by: Natasja Nortier <natasja.nortier@alliander.com>
